portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,7 +27,19 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    ArrayList<String> greetings = new ArrayList<String>();
+    greetings.add("Hello Tayyaba!");
+    greetings.add("Hello User!");
+    greetings.add("Hello World!");
+    String json = convertToJsonUsingGson(greetings);
     response.setContentType("text/html;");
-    response.getWriter().println("Hello Tayyaba!");
+    response.getWriter().println(json);
+  }
+
+  private String convertToJsonUsingGson(ArrayList array) {
+    Gson gson = new Gson();
```

Comment:
Consider having a private field of Gson in the DataServlet class, e.g.:
  private final Gson gson;

The reason is that it makes it possible for future member method like doPost() to reuse the same Gson instance.